service/controller/v14/resource/service/resource.go

```diff
@@ -81,3 +112,26 @@ func toServices(v interface{}) ([]*apiv1.Service, error) {
 
 	return services, nil
 }
+
+// portsEqual is a function that is checking if ports in the service have same important values.
+func portsEqual(a, b *apiv1.Service) bool {
```

Comment:
I would like to see some unit tests for this one as well.

Reply:
there is already test for it in the `isServiceModified` as this one is used there, does it make sens to copy the test case for this function?

Reply:
Yes you have a test for the mismatch. The benefit of testing the individual function is if someone changes something in the function they won't have to search for the reason why a different test is failing. A unit test should make sure the function itself works as expected.

Additionally I don't believe that you have situations tested like one service doesn't have ports at all, there are no ports on either service and so on.

This is my opinion here, if you feel comfortable with this - go ahead.

Reply:
Yep, you were right, i added test fro that function too.